File: SpearoGo/Views/WaterPage.swift
import SwiftUI

struct WaterPage: View {
    @Environment(AppState.self) private var appState

    var body: some View {
        VStack(spacing: Brand.Spacing.section) {
            Text("Water")
                .brandSectionHeader()

            if let marine = appState.marineData {
                Grid(alignment: .center, horizontalSpacing: 20, verticalSpacing: Brand.Spacing.section) {
                    GridRow {
                        ConditionItem(icon: "thermometer.medium",
                                      label: "Temp",
                                      value: String(format: "%.0f", marine.seaSurfaceTemp),
                                      unit: "°C")
                        ConditionItem(icon: "eye",
                                      label: "Viz",
                                      value: vizLabel(marine.waveHeight),
                                      unit: "")
                    }
                }
                .accessibilityElement(children: .combine)
                .accessibilityLabel(waterAccessibilityLabel(marine: marine))

                Text(wetsuitTip(marine.seaSurfaceTemp))
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, Brand.Spacing.page)
                    .infoPill()
                    .accessibilityLabel("Wetsuit tip: \(wetsuitTip(marine.seaSurfaceTemp))")
            } else {
                // Shimmer skeleton
                Grid(alignment: .center, horizontalSpacing: 20, verticalSpacing: Brand.Spacing.section) {
                    GridRow {
                        ConditionItemSkeleton()
                        ConditionItemSkeleton()
                    }
                }
                SkeletonBlock(width: 120, height: 24)
                    .accessibilityLabel("Loading water conditions")
            }
        }
        .padding(Brand.Spacing.page)
        .brandPage()
    }

    private func vizLabel(_ waveHeight: Double) -> String {
        switch waveHeight {
        case ..<0.5: return "Great"
        case 0.5..<1: return "Good"
        case 1..<1.5: return "Fair"
        default:     return "Poor"
        }
    }

    private func wetsuitTip(_ temp: Double) -> String {
        switch temp {
        case ..<15: return "Cold. 7mm + gloves."
        case 15..<20: return "Chilly. 5mm wetsuit."
        case 20..<25: return "Comfortable. 3mm."
        default:      return "Warm. 1–2mm or skin."
        }
    }

    private func waterAccessibilityLabel(marine: MarineData) -> String {
        let temp = String(format: "Water temperature %.0f degrees celsius", marine.seaSurfaceTemp)
        let viz = "Visibility \(vizLabel(marine.waveHeight))"
        return "\(temp). \(viz)"
    }
}

// MARK: - Previews

#Preview {
    WaterPage()
        .previewAsWatch()
        .environment(AppState.preview())
}

#Preview("Loading") {
    WaterPage()
        .previewAsWatch()
        .environment(AppState.previewLoading())
}

File: wear/app/src/main/java/com/spearotracker/spearogo/ui/pages/WaterPage.kt
package com.spearotracker.spearogo.ui.pages

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.wear.compose.material3.Text
import com.spearotracker.spearogo.ui.AppUiState
import com.spearotracker.spearogo.ui.components.ConditionItem
import com.spearotracker.spearogo.ui.components.ConditionItemSkeleton
import com.spearotracker.spearogo.ui.theme.Brand

@Composable
fun WaterPage(uiState: AppUiState) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(Brand.Spacing.page),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Text(
            text = "WATER",
            style = Brand.Typography.sectionHeader,
            color = Brand.Colors.textSecondary,
            modifier = Modifier.padding(bottom = Brand.Spacing.item)
        )

        val marine = uiState.marineData

        if (marine != null) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(20.dp),
                modifier = Modifier.padding(bottom = Brand.Spacing.section)
            ) {
                ConditionItem(icon = "temp", label = "Temp", value = "%.0f".format(marine.seaSurfaceTemp), unit = "\u00B0C")
                ConditionItem(icon = "eye", label = "Viz", value = vizLabel(marine.waveHeight), unit = "")
            }

            // Wetsuit tip
            Box(
                modifier = Modifier
                    .background(
                        Brand.Colors.secondary.copy(alpha = 0.08f),
                        RoundedCornerShape(Brand.Radius.chip)
                    )
                    .border(
                        1.dp,
                        Brand.Colors.secondary.copy(alpha = 0.15f),
                        RoundedCornerShape(Brand.Radius.chip)
                    )
                    .padding(horizontal = 8.dp, vertical = 5.dp)
            ) {
                Text(
                    text = wetsuitTip(marine.seaSurfaceTemp),
                    style = Brand.Typography.caption,
                    color = Brand.Colors.secondary,
                    textAlign = TextAlign.Center
                )
            }
        } else {
            Row(horizontalArrangement = Arrangement.spacedBy(20.dp), modifier = Modifier.padding(bottom = Brand.Spacing.section)) {
                ConditionItemSkeleton()
                ConditionItemSkeleton()
            }
        }
    }
}

private fun vizLabel(waveHeight: Double): String = when {
    waveHeight < 0.5 -> "Great"
    waveHeight < 1.0 -> "Good"
    waveHeight < 1.5 -> "Fair"
    else -> "Poor"
}

private fun wetsuitTip(temp: Double): String = when {
    temp < 15 -> "Cold. 7mm + gloves."
    temp < 20 -> "Chilly. 5mm wetsuit."
    temp < 25 -> "Comfortable. 3mm."
    else -> "Warm. 1-2mm or skin."
}
